### ISSUE TITLE
[Go] Building a record that contains a struct with a boolean field fails

### ISSUE THREAD

**Issue by coltonmorris** [apache/arrow#4239]
When I build a record that has a struct, and that struct has a field with the type `arrow.FixedWidthTypes.Boolean`, it later panics when trying to access the boolean value with the following error:
```
    runtime error: invalid memory address or nil pointer dereference
```

The following code successfully reads the string value, but not the boolean value:
```go
mem := memory.NewGoAllocator()

fields := []arrow.Field{
    {Name: "my struct", Type: arrow.StructOf([]arrow.Field{
        {Name: "my string", Type: arrow.BinaryTypes.String},
        {Name: "my bool", Type: arrow.FixedWidthTypes.Boolean},
    }...)},
}

schema := arrow.NewSchema(fields, nil)

bld := array.NewRecordBuilder(mem, schema)
defer bld.Release()

sb := bld.Field(0).(*array.StructBuilder)
f1b := sb.FieldBuilder(0).(*array.StringBuilder)
f2b := sb.FieldBuilder(1).(*array.BooleanBuilder)

sb.AppendValues([]bool{true})
f1b.AppendValues([]string{"someone", "was", "here"}, nil)
f2b.AppendValues([]bool{false, true, true}, nil)

r := bld.NewRecord()

struc := r.Column(0).(*array.Struct)

Expect(struc.Field(0).(*array.String).Value(0)).ToNot(Panic())
Expect(struc.Field(1).(*array.Boolean).Value(0)).To(Panic())
```

**Comment by emkornfield** [comment]
Could you file a [JIRA issue](https://issues.apache.org/jira/).  I'm not familiar with the Go code but this looks like a bug.

CC @sbinet

**Comment by sbinet** [comment]
I believe this has been fixed by #4229 which is now the new `master`.
(feel free to reopen if that's not the case on your end.)
